build_runner_core/pubspec.yaml

```diff
@@ -22,6 +22,7 @@ dependencies:
   meta: ^1.1.0
   path: ^1.1.0
   pool: ^1.0.0
+  pub_semver: ^1.4.1
```

Comment:
We are trying to keep build_runner_core as lightweight as possible. Is this package actually necessary? Can we do simple string comparison instead given the use case?